app/src/main/java/com/example/mycalculator/MainActivity.java

```diff
@@ -2,13 +2,99 @@
+    }
+
+    private void setListeners() {
+        binding.buttonNumberZero.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                presenter.numberPressed((String) binding.buttonNumberZero.getText().toString());
+            }
+        });
+        binding.buttonNumberOne.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                presenter.numberPressed((String) binding.buttonNumberOne.getText().toString());
+            }
+        });
+        binding.buttonNumberTwo.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                presenter.numberPressed((String) binding.buttonNumberTwo.getText().toString());
+            }
+        });
+        binding.buttonNumberThree.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                presenter.numberPressed((String) binding.buttonNumberThree.getText().toString());
+            }
+        });
+        binding.buttonNumberFour.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                presenter.numberPressed((String) binding.buttonNumberFour.getText().toString());
+            }
+        });
+        binding.buttonNumberFive.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                presenter.numberPressed((String) binding.buttonNumberFive.getText().toString());
+            }
+        });
+        binding.buttonNumberSix.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                presenter.numberPressed((String) binding.buttonNumberSix.getText().toString());
+            }
+        });
+        binding.buttonNumberSeven.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                presenter.numberPressed((String) binding.buttonNumberSeven.getText().toString());
+            }
+        });
+        binding.buttonNumberEight.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                presenter.numberPressed((String) binding.buttonNumberEight.getText().toString());
+            }
+        });
+        binding.buttonNumberNine.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                presenter.numberPressed((String) binding.buttonNumberNine.getText().toString());
+            }
+        });
+        binding.buttonFunctionErase.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                presenter.numberPressed("");
```

Comment:
no hardcode pls, create a StringUtils class and put there a EMPTY_STRING constant

Reply:
Done. Thx!!